data/tutorials/getting-started/2_00_editor_setup.md

```diff
@@ -49,10 +49,74 @@ If you used the DkML distribution, you will need to:
     1. Go to `File` > `Preferences` > `Settings` view (or press `Ctrl ,`)
     2. Select `User` > `Extensions` > `OCaml Platform`
     3. Uncheck `OCaml: Use OCaml Env`. That's it!
+    
+## Emacs
 
-## Vim and Emacs
+Using Emacs as the main OCaml editor requires at least two modes:
 
-**For Vim and Emacs**, we won't use the LSP server but rather directly talk to Merlin.
+- A major mode, which, among other things, supports syntax highlighting and the structuring of indentation levels
+- A minor mode, which will interact with a server language (such as `ocaml-lsp-server` or `merlin`). In this tutorial, we will focus on using `ocaml-lsp-server` as a server.
+
+### Choosing a major mode
+
+There are several major modes dedicated to OCaml, although there are 3 main ones:
+
+- [Tuareg](https://github.com/ocaml/tuareg): an old-fashioned (but still updated), very complete method, often recommended
+- [Caml](https://github.com/ocaml/caml-mode): a mode even older than `tuareg` (but still updated), lighter than `tuareg`
+- [Neocaml](https://github.com/bbatsov/neocaml): a brand new mode, based on modern tools (like [tree-sitter](https://tree-sitter.github.io/tree-sitter/)). Currently still very experimental!
+
+For the purposes of this tutorial, we're going to focus on the use of `tuareg` as the main mode, but that's a detail and you're free to experiment and choose your favourite mode! You can add these lines to your Emacs configuration:
+
+```elisp
+(use-package tuareg
+  :ensure t
+  :mode (("\\.ocamlinit\\'" . tuareg-mode)))
+```
+
+> If your version of Emacs does not support the `use-package` macro (or is not set up to take MELPA packages into account), please update it and follow these instructions for [`use-package`](https://github.com/jwiegley/use-package) and to [add MELPA](https://melpa.org/#/getting-started).
+
+### LSP setup for OCaml
+
+Since version `29.1`, Emacs has had a mode for interacting with an LSP server, [Eglot](https://www.gnu.org/software/emacs/manual/html_mono/eglot.html). If you are using an earlier version of Emacs, you will need to install it this way:
```

Comment:
```suggestion
Since version `29.1`, Emacs has had a built-in mode for interacting with LSP servers, [Eglot](https://www.gnu.org/software/emacs/manual/html_mono/eglot.html). If you are using an earlier version of Emacs, you will need to install it this way:
```